src/System.Text.Utf8String/System/Text/Utf8String.cs

```diff
@@ -11,29 +11,48 @@
 namespace System.Text.Utf8
 {
     [DebuggerDisplay("{ToString()}u8")]
-    public partial ref struct Utf8String
+    public partial class Utf8String
     {
-        private readonly ReadOnlySpan<byte> _buffer;
+        private readonly byte[] _buffer;
 
         private const int StringNotFound = -1;
 
-        static Utf8String s_empty => default;
+        static Utf8String s_empty = new Utf8String(string.Empty);
 
         // TODO: Validate constructors, When should we copy? When should we just use the underlying array?
         // TODO: Should we be immutable/readonly?
         public Utf8String(ReadOnlySpan<byte> buffer)
         {
-            _buffer = buffer;
+            _buffer = buffer.ToArray();
         }
 
         public Utf8String(byte[] utf8bytes)
         {
-            _buffer = new ReadOnlySpan<byte>(utf8bytes);
+            _buffer = utf8bytes;
         }
 
         public Utf8String(byte[] utf8bytes, int index, int length)
         {
-            _buffer = new ReadOnlySpan<byte>(utf8bytes, index, length);
+            if (index < 0)
+            {
+                throw new ArgumentOutOfRangeException("index");
+            }
+
+            if (length < 0)
+            {
+                // TODO: Should we support that?
+                throw new ArgumentOutOfRangeException("length");
+            }
+
+            if (index + length > utf8bytes.Length)
+            {
+                throw new ArgumentOutOfRangeException("index");
+            }
+
+            // Length checking
+            _buffer = new byte[length];
+            for (int i = 0; i < length; i++)
+                _buffer[i] = utf8bytes[i + index];
```

Comment:
It's strange that this ctor would copy, but the one taking byte[] would not. 

Also, the code should be written as follows:
```c#
_buffer = new Span<byte>(utf8Bytes, index, length).ToArray();
```